### PR TITLE
[trivial] minor naming fixes

### PR DESCRIPTION
**Pull request by georgemitenkov**
## Description

While going over Move repo code, there are multiple instances where things are just called confusingly or wrong. This PR fixes that, in the hope that certain coding conventions and better code review practices can be established to mitigate such changes in the future.

## Type of Change
- [ ] New feature
- [ ] Bug fix
- [ ] Breaking change
- [ ] Performance improvement
- [x] Refactoring
- [ ] Dependency update
- [ ] Documentation update
- [ ] Tests

## Which Components or Systems Does This Change Impact?
- [ ] Validator Node
- [ ] Full Node (API, Indexer, etc.)
- [x] Move/Aptos Virtual Machine
- [ ] Aptos Framework
- [ ] Aptos CLI/SDK
- [ ] Developer Infrastructure
- [ ] Other (specify)

## How Has This Been Tested?

Replay: https://github.com/aptos-labs/aptos-core/actions/runs/9081435243

## Checklist
- [x] I have read and followed the [CONTRIBUTING](https://github.com/aptos-labs/aptos-core/blob/main/CONTRIBUTING.md) doc
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I identified and added all stakeholders and component owners affected by this change as reviewers
- [ ] I tested both happy and unhappy path of the functionality
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: third_party/move/move-vm/runtime/src/loader/function.rs [hunk 1]
```diff
@@ -37,16 +37,16 @@ pub(crate) struct Function {
...
     pub(crate) def_is_native: bool,
     pub(crate) def_is_friend_or_private: bool,
     pub(crate) scope: Scope,
     pub(crate) name: Identifier,
-    pub(crate) return_types: Vec<Type>,
-    pub(crate) local_types: Vec<Type>,
-    pub(crate) parameter_types: Vec<Type>,
+    pub(crate) return_tys: Vec<Type>,
+    pub(crate) local_tys: Vec<Type>,
+    pub(crate) arg_tys: Vec<Type>,
```
Comment (vgao1996) [inline third_party/move/move-vm/runtime/src/loader/function.rs:49]: The `types -> ty` changes look good and I would be open to making it a convention.

The `parameter -> arg` change though, might need to be revisited. I could be wrong, but my impression is that it is a commonly accepted convention within the PL space that
- "Parameters" are used to denote value bindings in function signatures -- in other words, a prototype of the value to be accepted
  - For example, `x` in `fun foo(x: u64)` is a parameter
- "Arguments" are used to denote actual values that get passed to a function
  - For example, `3` in `foo(3)` is an argument

I don't recall any authoritative definitions of these terms, but if you google "function parameters vs arguments" you'll see many people defining them this way.

Comment (vgao1996) [reply third_party/move/move-vm/runtime/src/loader/function.rs:49]: We actually a discussion on this back in the old days, and that resulted in us changing some of the stuff to `type formals/type actuals`. While I personally like those names for their preciseness, they are probably a bit too formal for anyone except maybe the most fervent PL people. We might want to consider changing those to `params/tys` instead.

Comment (vgao1996) [reply third_party/move/move-vm/runtime/src/loader/function.rs:49]: There is one misnaming I know though
```
pub struct StructTag {
    pub address: AccountAddress,
    pub module: Identifier,
    pub name: Identifier,
    // alias for compatibility with old json serialized data.
    #[serde(rename = "type_args", alias = "type_params")]
    pub type_params: Vec<TypeTag>,
}
```
Here the `TypeTag`s are actual arguments to the type constructor, so they should be called arguments instead of parameters.

**Review by vgao1996 — COMMENTED** [review]
Love to see cleanups like this coming up!

File: third_party/move/move-vm/runtime/src/loader/function.rs (hunk 1 above)
Comment (georgemitenkov) [reply third_party/move/move-vm/runtime/src/loader/function.rs:49]: Yes, we have `params` and `args` used interchangeably, I would love us to just use one convention, no strong opinion which one. Let me use `params` then where possible 👍

Comment (georgemitenkov) [reply third_party/move/move-vm/runtime/src/loader/function.rs:49]: I think being less informal with `params/tys` is better if we want to attract external contributors 😄

Comment (vgao1996) [reply third_party/move/move-vm/runtime/src/loader/function.rs:49]: > we have params and args used interchangeably

Only in some places, and those are definitely cases of misnaming. The rest of the code follows either the "parameter/argument" or the "formal/actual" conventions.

**Comment by georgemitenkov** [comment]
@vgao1996 updated naming based on your suggestions.

File: third_party/move/move-vm/runtime/src/loader/script.rs [hunk 2]
```diff
@@ -124,30 +118,24 @@ impl Script {
...
             def_is_friend_or_private: false,
             scope,
             name,
-            return_types: return_tys.clone(),
-            local_types: local_tys,
-            parameter_types: parameter_tys.clone(),
+            // Script must not return values.
+            return_tys: vec![],
+            local_tys,
+            param_tys,
```
Comment (vgao1996) [inline third_party/move/move-vm/runtime/src/loader/script.rs:138]: Note: this should be fine as scripts cannot return values for now. We'll know to come back to this if something ends up getting broken.

File: third_party/move/move-vm/runtime/src/loader/mod.rs [hunk 3]
```diff
@@ -291,7 +291,7 @@ impl Loader {
     pub(crate) fn load_script(
         &self,
         script_blob: &[u8],
-        ty_args: &[TypeTag],
+        ty_arg_tags: &[TypeTag],
```
Comment (vgao1996) [inline third_party/move/move-vm/runtime/src/loader/mod.rs:294]: Probably not worth to emphasize the fact that these args are represented as tags here.

File: third_party/move/move-vm/runtime/src/runtime.rs [hunk 4]
```diff
@@ -304,13 +305,7 @@ impl VMRuntime {
...
         let (layout, has_identifier_mappings) = self
             .loader
-            .type_to_type_layout_with_identifier_mappings(ty, module_store)
-            .map_err(|_err| {
-                // TODO: Should we use `err` instead of mapping?
-                PartialVMError::new(StatusCode::VERIFICATION_ERROR).with_message(
-                    "entry point functions cannot have non-serializable return types".to_string(),
-                )
-            })?;
+            .type_to_type_layout_with_identifier_mappings(ty, module_store)?;
```
Comment (vgao1996) [inline third_party/move/move-vm/runtime/src/runtime.rs:308]: This either needs to be
1. Feature-gated
2. Reverted (better to do it in a separate PR)

Let's do (2) as we discussed offline

**Review by vgao1996 — APPROVED** [review]
Might also want to consider renaming "type_formals" to "type_params" and "type_actuals" to "type_args", but no need to include them in the current PR.

File: third_party/move/move-vm/runtime/src/runtime.rs (hunk 4 above)
Comment (georgemitenkov) [reply third_party/move/move-vm/runtime/src/runtime.rs:308]: Fixed

File: third_party/move/move-vm/runtime/src/loader/mod.rs (hunk 3 above)
Comment (georgemitenkov) [reply third_party/move/move-vm/runtime/src/loader/mod.rs]: Changed back to types.

**Review by runtian-zhou — APPROVED** [review]
